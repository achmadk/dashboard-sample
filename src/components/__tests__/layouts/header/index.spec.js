import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import chai from 'chai'

import HeaderContainer, { headerComponent } from '@/components/layouts/header'
import HeaderComponent from '@/components/layouts/header/header'
import ProviderStore from '@/../test/provider-store'

describe('header Layout Container', () => {
  it('render correctly', () => {
    const wrapper = shallow(<HeaderContainer />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('return layout component successfully', () => {
    const result = headerComponent()
    chai.expect(result).to.exist
  })
})

describe('header Layout Component', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <ProviderStore>
        <HeaderComponent />
      </ProviderStore>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
