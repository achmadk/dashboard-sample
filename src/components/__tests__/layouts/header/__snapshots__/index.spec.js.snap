// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header Layout Component render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectFunction />
</Provider>
`;

exports[`header Layout Container render correctly 1`] = `
<React.Suspense
  fallback={
    <div>
      Loading...
    </div>
  }
>
  <React.Lazy />
</React.Suspense>
`;
